extensions/sql-database-projects/src/tools/buildHelper.ts

```diff
@@ -37,11 +42,14 @@ export class BuildHelper {
 		const extName = utils.getAzdataApi() ? sqldbproj.extension.name : sqldbproj.extension.vsCodeName;
 		this.extensionDir = vscode.extensions.getExtension(extName)?.extensionPath ?? '';
 		this.extensionBuildDir = path.join(this.extensionDir, buildDirectory);
+
 	}
 
-	// create build dlls directory
-	// this should not be required. temporary solution for issue #10273
-	public async createBuildDirFolder(): Promise<void> {
+	/**
+	 * Create build dlls directory with the dlls and targets needed for building a sqlproj
+	 * @param outputChannel
+	 */
+	public async createBuildDirFolder(outputChannel?: vscode.OutputChannel): Promise<void> {
```

Comment:
Why is outputChannel optional? There should always be an output channel (or a mocked one during tests)

Reply:
yup will fix this before changing it from draft. It was optional so that the tests wouldn't complain while I was doing this initial POC

Reply:
fixed